src/server/templates/typescript.ts

```diff
@@ -390,11 +392,18 @@ export type Database = {
     })}
 }
 
-type PublicSchema = Database[Extract<keyof Database, "public">]
+${
+  defaultSchema
+    ? `export type DefaultSchema = Database[Extract<keyof Database, ${JSON.stringify(defaultSchema)}>]
+  export type DefaultSchemaOrPublic = DefaultSchema`
+    : `export type PublicSchema = Database[Extract<keyof Database, "public">]
+  export type DefaultSchemaOrPublic = PublicSchema`
+}
+
 
 export type Tables<
   PublicTableNameOrOptions extends
-    | keyof (PublicSchema["Tables"] & PublicSchema["Views"])
+    | keyof (DefaultSchemaOrPublic["Tables"] & DefaultSchemaOrPublic["Views"])
```

Comment:
Better to use `DefaultSchema` nomenclature everywhere vs. `PublicSchema`

Reply:
Resolved